library/core/src/future/future.rs

```diff
@@ -5,7 +5,7 @@ use crate::ops;
 use crate::pin::Pin;
 use crate::task::{Context, Poll};
 
-/// A future represents an asynchronous computation.
+/// A future represents an asynchronous computation obtained by use of [`async`].
```

Comment:
This might be confusing; it isn't only obtained via `async` since Future is implemented manually for e.g. system resources. Most `await` chains bottom out in one of these manual implementations.

Reply:
Feel free to follow up with another PR.